### PR TITLE
Repeated dependencies in package.json

### PR DESCRIPTION
**Pull request by carloshpds**
@jackfranklin, #93 - nice error for repeated dependencies in package.json implemented ;)

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -71,17 +71,22 @@ module.exports = function(options) {
...
+            return requireFn(name);
+          }
+        });
+      } else {
+        logDebug('requiring ' + name + '...');
+        object[requireName] = requireFn(name);
+      }
+    } catch(err){
+      logDebug('error: defineProperty ' + name);
+      throw new Error('Could not define the property \"' + requireName + '\", you may have repeated dependencies in your package.json like' + ' "gulp-' + requireName + '" and ' + '"' + requireName + '"' );
```
Comment (callumacrae) [inline index.js]: Would it not be better to just check whether the property exists already? I can see this catching other unwanted stuff.

Comment (jackfranklin) [inline index.js]: I'd agree with that - just check when going to define the property if it already exists first.

File: index.js [hunk 2]
```diff
@@ -102,7 +107,11 @@ module.exports = function(options) {
   var scopeTest = new RegExp('^@');
   var scopeDecomposition = new RegExp('^@(.+)/(.+)');
 
-  multimatch(names, pattern).forEach(function(name) {
+  var multimatched = multimatch(names, pattern);
+  logDebug('multimatch pattern: ' + pattern);
+  logDebug('multimatched: ' + multimatched);
```
Comment (jackfranklin) [inline index.js]: can we remove these please :)

File: index.js [hunk 3]
```diff
@@ -110,8 +119,10 @@ module.exports = function(options) {
         finalObject[decomposition[1]] = {};
       }
 
+      logDebug('scopeTest returned true for: ' + name);
       defineProperty(finalObject[decomposition[1]], getRequireName(decomposition[2]), name);
     } else {
+      logDebug('scopeTest returned false for: ' + name);
```
Comment (jackfranklin) [inline index.js]: similarly I don't think the scopeTest debug logs will be that helpful generally

Comment (callumacrae) [inline index.js]: From the looks of things, all the changes after line 100 in this file are safe to be reverted. No logic has actually changed.

File: index.js (hunk 2 above)
Comment (carloshpds) [inline index.js]: Yeah! But it was very helpful to understand what was happing, cause the error occurs only if you change the default pattern.
Anyway, I will remove this logs ;)

File: index.js (hunk 1 above)
Comment (carloshpds) [inline index.js]: Agree, it would be better. I'll change it.

Comment (carloshpds) [inline index.js]: Do you think that the line ` logDebug('error: defineProperty ' + name);` is really helpful ?

File: index.js (hunk 3 above)
Comment (carloshpds) [inline index.js]: Done ;)

File: index.js [hunk 4]
```diff
@@ -112,6 +117,7 @@ module.exports = function(options) {
 
       defineProperty(finalObject[decomposition[1]], getRequireName(decomposition[2]), name);
     } else {
+
```
Comment (jackfranklin) [inline index.js]: can we remove this superfluous line please?

**Comment by jackfranklin** [comment]
@carloshpds awesome! One more teeny pedantic comment but great work :)

File: index.js (hunk 4 above)
Comment (carloshpds) [inline index.js]: I knew it !! Hahahaha ..it was bothering me too

Comment (carloshpds) [inline index.js]: Done ;)

**Comment by carloshpds** [comment]
Thanks @jackfranklin :)

**Comment by jackfranklin** [comment]
@carloshpds I just pushed 1.2.0, which has this in it! Thanks again!

**Comment by carloshpds** [comment]
:smile: :+1:

**Pull request merged**